sync/sync_test.go

```diff
@@ -41,7 +42,12 @@ func TestSyncBlocks(t *testing.T) {
 		testDB := db.NewTestDb()
 		bc := blockchain.NewBlockchain(testDB, utils.MAINNET)
 		synchronizer := NewSynchronizer(bc, gw, log)
-		assert.Error(t, synchronizer.SyncBlocks())
+		ctx, cancel := context.WithCancel(context.Background())
+		go func() {
+			time.Sleep(time.Second)
+			cancel()
+		}()
+		assert.EqualError(t, synchronizer.Run(ctx), "context canceled")
```

Comment:
We never return context cancelled error, so how does this make sense? Also, we should use `errors.Is()/As()` for error comparisons

Reply:
Yeap, it doesnt make sense:) fixed